src/utils/roll-orb.ts

```diff
@@ -0,0 +1,104 @@
+import * as debug from 'debug';
+import * as jsyaml from 'js-yaml';
+
+import { ORBS, OrbTarget } from '../constants';
+import { getOctokit } from './octokit';
+import { getOrbPRText } from './pr-text-orb';
+
+export interface RollOrbParams {
+  orbTarget: OrbTarget;
+  electronBranch;
+  targetValue: string;
+  repository: {
+    owner: string;
+    repo: string;
+  };
+}
+
+// Rolls an orb in a .circleci/config.yml file to a new version
+export async function rollOrb({
+  orbTarget: rollTarget,
+  electronBranch,
+  targetValue,
+  repository,
+}: RollOrbParams): Promise<any> {
+  const d = debug(`roller/${rollTarget.name}:rollOrb()`);
+  const github = await getOctokit();
+
+  try {
+    d(`roll triggered for  ${rollTarget.name}=${targetValue}`);
+
+    const sha = electronBranch.commit.sha;
+    const filePath = '.circleci/config.yml';
+    const branchName = `roller/${rollTarget.name}/${electronBranch.name}`;
+    const shortRef = `heads/${branchName}`;
+    const ref = `refs/${shortRef}`;
+    const { owner, repo } = repository;
+
+    const response = await github.repos.getContent({
+      owner,
+      repo,
+      path: filePath,
+    });
+
+    if ('type' in response.data && 'content' in response.data && response.data.type == 'file') {
+      const content = Buffer.from(response.data.content, 'base64').toString();
+      const yamlData = jsyaml.load(content);
+      let curr = yamlData[ORBS];
+
+      // attempt to find the orb in .circleci/config.yml whos value includes `orbTarget.name`
+      let targetKey;
+      for (const key of Object.keys(curr)) {
+        if (curr[key].includes(rollTarget.name)) {
+          targetKey = key;
+          break;
+        }
+      }
+
+      if (targetKey === undefined) {
+        d(`Key for ${rollTarget.name} not found.`);
+        throw new Error(`Key for "${rollTarget.name}" not found.`);
```

Comment:
Is this an error? Shouldn't this simply skip CircleCI configs which don't use the orb?

Reply:
The roller is called by the handler, which pulls a list of repos with valid CircleCi configs- if it gets to this point and still doesn't have a key, it would be an error

That being said, this was left over from before a refactor, and now serves as a potentially vestigial check. Happy to remove it if requested

Reply:
Where? `getRelevantReposList` only checks that a CircleCI config exists - not that it's using the target orb. Then this code will throw an error if the key isn't there.

Reply:
oops, ur totally right, head-canon mishap on my end :P 

It should skip if there's no key